plugins/module_utils/foreman_helper.py

```diff
@@ -120,27 +120,38 @@ def list_resource(self, resource, search, params=None):
         params['per_page'] = 2 << 31
         return self.foremanapi.resource(resource).call('index', params)['results']
 
-    def find_resource(self, resource, search, params=None, failsafe=False, thin=False):
+    def find_resource(self, resource, search='', name=None, params=None, failsafe=False, thin=False):
+        if name is not None:
+            search += 'name="{}"'.format(name)
```

Comment:
```suggestion
            search += ' name="{}"'.format(name)
```
the space is important, because if `search` already contained something, you'd end up with `the_old_searchname="blah"` otherwise

Reply:
I can add the space  but then all other modules have to be re-recorded else the new addition will fail test for them because **strip** function does not modify the String it returns a new one.